Sources/Build/BuildPlan.swift

```diff
@@ -1298,7 +1298,8 @@ public class BuildPlan {
             let swiftTarget = SwiftTarget(
                 testDiscoverySrc: src,
                 name: testProduct.name,
-                dependencies: testProduct.underlyingProduct.targets.map { .target($0, conditions: []) }
+                dependencies: testProduct.underlyingProduct.targets.map { .target($0, conditions: []) },
+                toolsVersion: .currentToolsVersion
```

Comment:
It's doubtful whether the current tools version is the right one for the targets created through test discovery. A better choice might be to set it to the version of the target being tested. Are were guaranteed at this point that all the targets in the array come from the same package?